bin/ovs-offline

```diff
@@ -324,21 +327,37 @@ do_collect-sos-ovs() {
 
     save_dir=${WORKDIR}/restore_flows
     mkdir -p $save_dir
+    
+    if [[ -n ${OVS_RUNDIR} && -f $imported_db_file ]]; then
+        # The DB already exists, we don't have to check the sosreport for the bridges/tlv-map/groups, etc. Some older versions of `sos` do not collect this.
+        # We do still need the flows from the sosreport, which should be present in most recent versions of sos (tested >= 3.8).
+        # If OVS_RUNDIR is present in the environment variables, then it means we have sourced the commands, so we can use `ovs-*` commands directly in the shell.
+        bridges=$(ovs-vsctl -t 5 list-br)
+        for br in $bridges; do
+            # Get the highest level of supported protocol for each bridge.
+            local high_ver=$(ovs-vsctl get Bridge $br protocols | tr -d '[|]' | awk '{print $NF}')
+            ovs-ofctl -O $high_ver dump-tlv-map $br > $save_dir/ovs-ofctl_-O_${high_ver}_dump-tlv-map_${br} || error "Error in executing the ovs-ofctl dump-tlv-map command. Please check your DB file."
+            ovs-ofctl -O $high_ver dump-groups $br > $save_dir/ovs-ofctl_-O_${high_ver}_dump-groups_${br} || error "Error in executing the ovs-ofctl dump-groups command. Please check your DB file."
 
-    if test -f "${SOS_CMD_DIR}/ovs-vsctl_-t_5_list-br"; then
-        bridges=$(cat ${SOS_CMD_DIR}/ovs-vsctl_-t_5_list-br)
+        done
     else
-        error "ovs-vsctl_-t_5_list-br missing from sos report. Make sure your sos archive is generated from the most recent sos report"
+        if test -f "${SOS_CMD_DIR}/ovs-vsctl_-t_5_list-br"; then
+            bridges=$(cat ${SOS_CMD_DIR}/ovs-vsctl_-t_5_list-br)
+        else
+            error "ovs-vsctl_-t_5_list-br missing from sos report. Make sure your sos archive is generated from the most recent sos report"
+        fi
+
+        # Collect group, flow, and tlv dumps
+        for br in $bridges; do
+            local version=$(ls $SOS_CMD_DIR | grep ovs-ofctl_-O_OpenFlow[0-9]*_dump-flows_${br} | grep -o "OpenFlow[0-9]*")
+            local high_ver=$(echo "$version" | sort | tail -n 1)
+            cp ${SOS_CMD_DIR}/ovs-ofctl_-O_${high_ver}_dump-tlv-map_${br} $save_dir || error "Sos report missing dump-tlv for bridge $br. Make sure your sos archive is generated from the most recent sos report"
+            cp ${SOS_CMD_DIR}/ovs-ofctl_-O_${high_ver}_dump-groups_${br} $save_dir || error "Sos report missing dump-groups for bridge $br. Make sure your sos archive is generated from the most recent sos report"
```

Comment:
Maybe a simpler approach  would be not to fail here:
We could just warn that groups ant tlv files are missing and that, if flows use them, flow insertion will fail and we cannot really do anything  about it.